sktime/classification/shapelet_based/_rocket_classifier.py

```diff
@@ -102,37 +123,41 @@ def fit(self, X, y):
         self : object
         """
         X, y = check_X_y(X, y)
+        n_jobs = check_n_jobs(self.n_jobs)
 
         self.n_classes = np.unique(y).shape[0]
         self.classes_ = class_distribution(np.asarray(y).reshape(-1, 1))[0][0]
-        for index, classVal in enumerate(self.classes_):
-            self.class_dictionary[classVal] = index
-
-        if self.ensemble:
-            for i in range(self.ensemble_size):
-                rocket_pipeline = make_pipeline(
-                    Rocket(
-                        num_kernels=self.num_kernels, random_state=self.random_state
-                    ),
-                    RidgeClassifierCV(alphas=np.logspace(-3, 3, 10), normalize=True),
+        for index, class_val in enumerate(self.classes_):
+            self.class_dictionary[class_val] = index
+
+        if self.n_estimators is not None:
+            if n_jobs is None:
```

Comment:
I don't think we need the if-else case distinction here, `n_jobs=None` is interpreted as `n_jobs=1` (unless inside a context manager)

Reply:
Yeah, that was previously the case (described here https://github.com/alan-turing-institute/sktime/pull/796#discussion_r612109920). Right now if `n_jobs` is not `None`, we fit a clone of the estimator rather than estimator itself. And random state is re-defined for all clones. Which  means that setting `n_jobs=1` be default would break reproducibility with previously trained classifiers. If it's okay to ignore this fact, I can re-work the code to remove one layer of if-else branches.

Reply:
I prefer fixing the bug here and have a single training loop with properly set random states (breaking reproducibility). Reproducibility can still be ensure by reference to the older sktime release. @TonyBagnall @ABostrom @jasonlines, any thoughts on this as code owners?

Reply:
Okay, I will revert to initializing `n_jobs` to 1

Reply:
Done!